src/provider/store/Store.ts

```diff
@@ -1,122 +1,92 @@
+        this._listeners.push(listener);
 
-    private watch<T>(getState: () => any, getObject: (state: RootState) => T): (observer: StoreChangeObserver<T>) => () => void {
-        let currentValue = getObject(getState());
-        return function w(observer: StoreChangeObserver<T>) {
-            return function () {
-                const newValue: T = getObject(getState());
-                if (currentValue !== newValue) {
-                    const oldValue = currentValue;
-                    currentValue = newValue;
-                    observer(oldValue, newValue);
-                }
-            };
+        return () => {
+            const index: number = this._listeners.indexOf(listener);
+            if (index >= 0) {
+                this._listeners.splice(index, 1);
+            }
         };
     }
 
-    private reduce<T extends Action>(state: RootState | undefined, action: ActionOf<T>): RootState {
-        const handler: ActionHandler<T> | undefined = this._actionHandlerMap[action.type] as ActionHandler<T>;
-
-        if (handler) {
-            return handler(state!, action);
-        } else {
-            // No handler registered for this action - action does not modify the store's state
-            return state!;
-        }
+    protected setState(state: S): void {
+        this._currentState = state;
     }
 
-    private createEnhancer(): StoreEnhancer {
-        let enhancer: StoreEnhancer = applyMiddleware(this.createMiddleware.bind(this));
-
-        if (process.env.NODE_ENV !== 'production') {
-            const devTools = composeWithDevTools({
-                // @ts-ignore: Property 'suppressConnectErrors' missing from type definitions
-                realtime: true, port: 9950, suppressConnectErrors: true, sendOnError: 2
-            });
-            enhancer = devTools(enhancer);
-        }
-
-        return enhancer;
+    private async reduceAndEmit(action: A) {
+        // emit signal last
+        this.reduce(action);
+        await this.onAction.emit(action);
     }
 
-    private createMiddleware(): (next: Dispatch<RootAction>) => (action: RootAction) => Promise<RootAction> {
-        return (next: Dispatch<RootAction>) => async (action: RootAction): Promise<RootAction> => {
-            await this.onAction.emit(action);
-
-            return next(action);
-        };
+    private reduce(action: A): void {
+        this._currentState = action.reduce(this.state);
+        this._listeners.forEach(listener => listener(() => this.getState()));
     }
 
-    private async getInitialState(): Promise<RootState> {
-        const notificationCollection: Collection<StoredNotification> = this._database.get(CollectionMap.NOTIFICATIONS);
-        const notifications: StoredNotification[] = await notificationCollection.getAll();
-
-        return Object.assign({}, Store.INITIAL_STATE, {notifications});
-    }
+    private replaceReducer(nextReducer: Reducer<S>): void {}
+    private [Symbol.observable](): void {}
```

Comment:
Typescript magic here that is beyond me. Able to give me an explanation?

Reply:
This can be removed, its from when we were trying to match the type of a "Store" with the definition that `<Provider store={...}>` takes. But since `any` is being removed this is not needed any more.

Reply:
removed